include/core/mir/geometry/dimensions.h

```diff
@@ -161,6 +169,20 @@ inline constexpr DeltaX operator*(DeltaX const& dx, Scalar scale) { return scale
 template<typename Scalar>
 inline constexpr DeltaY operator*(DeltaY const& dy, Scalar scale) { return scale*dy; }
 
+// Converting between types is fine, as long as they are along the same axis
+inline constexpr Width as_width(DeltaX const& dx) { return Width{dx.as_int()}; }
+inline constexpr Height as_height(DeltaY const& dy) { return Height{dy.as_int()}; }
+inline constexpr X as_x(DeltaX const& dx) { return X{dx.as_int()}; }
+inline constexpr Y as_y(DeltaY const& dy) { return Y{dy.as_int()}; }
+inline constexpr DeltaX as_delta(X const& x) { return DeltaX{x.as_int()}; }
+inline constexpr DeltaY as_delta(Y const& y) { return DeltaY{y.as_int()}; }
+inline constexpr X as_x(Width const& w) { return X{w.as_int()}; }
+inline constexpr Y as_y(Height const& h) { return Y{h.as_int()}; }
+inline constexpr Width as_width(X const& x) { return Width{x.as_int()}; }
+inline constexpr Height as_height(Y const& y) { return Height{y.as_int()}; }
+inline constexpr DeltaX as_delta(Width const& w) { return DeltaX{w.as_int()}; }
+inline constexpr DeltaY as_delta(Height const& h) { return DeltaY{h.as_int()}; }
```

Comment:
These ones are... ok?

I'd kinda prefer things using these to be eg: `{0} + delta;`, but C++ won't infer the right things there…

Reply:
The problem isn't time inference, the problem is that `auto left_margin = {0} + ((rect.size.width - {0}) - exclusive.left() - {0}))` is not easy code to write or to read.